jmx-scraper/src/main/java/io/opentelemetry/contrib/jmxscraper/JmxScraper.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.jmxscraper;
+
+import io.opentelemetry.contrib.jmxscraper.client.JmxRemoteClient;
+import io.opentelemetry.contrib.jmxscraper.config.ConfigurationException;
+import io.opentelemetry.contrib.jmxscraper.config.JmxScraperConfig;
+import io.opentelemetry.contrib.jmxscraper.config.JmxScraperConfigFactory;
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.Properties;
+import java.util.logging.Logger;
+import javax.management.MBeanServerConnection;
+import javax.management.remote.JMXConnector;
+
+public class JmxScraper {
+  private static final Logger logger = Logger.getLogger(JmxScraper.class.getName());
+
+  private final JmxRemoteClient client;
+
+  // TODO depend on instrumentation 2.9.0 snapshot
+  // private final JmxMetricInsight service;
+
+  /**
+   * Main method to create and run a {@link JmxScraper} instance.
+   *
+   * @param args - must be of the form "-config {jmx_config_path,'-'}"
+   */
+  @SuppressWarnings({"SystemOut", "SystemExitOutsideMain"})
+  public static void main(String[] args) {
+    JmxScraperConfig config;
+    JmxScraper jmxScraper = null;
+    try {
+      JmxScraperConfigFactory factory = new JmxScraperConfigFactory();
```

Comment:
This factory is only used inside of the `createConfigFromArgs` method, so it doesn't need to be created up here and passed in -- it can just be created in that method.